File: Modules/SEDO/EMS/SIE.Web.EMS.Purchases/EquipmentSetups/Commands/ReassignmentCommand.js
SIE.defineCommand('SIE.Web.EMS.Purchases.EquipmentSetups.Commands.ReassignmentCommand', {
    meta: { text: "转派", group: "edit", iconCls: "icon-EditEntity icon-blue" },
    canExecute: function (view) {
        var selectModels = view.getSelection();
        if (selectModels.length == 0) return false;
        var res = true;
        SIE.each(selectModels, function (model) {
            if (model.data.ApprovalStatus !== 10 && model.data.ApprovalStatus !== 50) {
                res = false;
                return false;
            }
        });
        return res;
    },
    execute: function (view, source) {
        SIE.AutoUI.getMeta({
            model: "SIE.EMS.Purchases.EquipmentSetups.ViewModels.SelPrincipalViewModel",
            ignoreCommands: true,
            isDetail: true,
            ignoreQuery: true,
            viewGroup: "DetailsView",
            callback: function (res) {
                var detailView = SIE.AutoUI.createDetailView(res);
                var entity = new detailView._model();
                detailView.setData(entity);
                var win = SIE.Window.show({
                    title: "选择员工".t(),
                    width: 480,
                    height: 200,
                    items: detailView.getControl(),
                    id: "ReassignmentCommand001",
                    callback: function (btn) {
                        if (btn == "确定".t()) {
                            var model = detailView.getData().data;
                            if (model.PrincipalId <= 0) {
                                SIE.Msg.showError("请选择员工".t());
                                return false;
                            }
                            var selectModels = view.getSelection();
                            var selectIds = view.getSelectionIds(selectModels);
                            view.execute({
                                withIds: true,
                                data: model.PrincipalId,
                                selectIds: selectIds,
                                success: function (resu) {
                                    SIE.Msg.showMessage("转派成功!".t());
                                    view.reloadData();
                                }
                            });
                            win.close();
                            return false;
                        }
                    }
                });
            }
        });
    }
});

File: Modules/SEDO/EMS/SIE.Web.EMS.Purchases/EquipmentSetups/Commands/ReassignmentCommand.cs
using SIE.EMS.Purchases.EquipmentSetups;
using SIE.Web.Command;
using System.Collections.Generic;

namespace SIE.Web.EMS.Purchases.EquipmentSetups.Commands
{
    /// <summary>
    /// 转派
    /// </summary>
    [JsCommand("SIE.Web.EMS.Purchases.EquipmentSetups.Commands.ReassignmentCommand")]
    public class ReassignmentCommand : ViewCommand
    {
        /// <summary>
        /// 执行命令
        /// </summary>
        /// <param name="args">参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            if (args == null)
            {
                return false;
            }
            List<double> selectedIds = new List<double>(args.SelectedIds);
            var principalId = args.Data.ToJsonObject<double>();
            RT.Service.Resolve<EquipmentSetupController>().Reassignment(selectedIds, principalId);
            return true;
        }
    }
}
